tests/30-main.spec.js

```diff
@@ -254,4 +255,450 @@ describe('main', () => {
+   'statusTypeMatches"', async () => {
+    let err;
+    let result;
+    try {
+      result = statusTypeMatches('{credential}');
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"credential" must be an object');
+  });
+
+  it('should fail to verify if credential is not an object for ' +
+    '"assertRevocationList2020Context"', async () => {
+    let err;
+    let result;
+    try {
+      result = assertRevocationList2020Context('{credential}');
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"credential" must be an object');
+  });
+
+  it('should fail to verify if credential is not an object for ' +
+    '"getCredentialStatus"', async () => {
+    let err;
+    let result;
+    try {
+      result = getCredentialStatus('{credential}');
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"credential" must be an object');
+  });
+
+  it('should fail to verify if @context is not an array in ' +
+    '"statusTypeMatches"', async () => {
+    const id = 'https://example.com/status/1';
+    const list = await createList({length: 100000});
+    const credential = await createCredential({id, list});
+    let err;
+    let result;
+    try {
+      // Changing the @context property to a string
+      credential['@context'] = 'https://example.com/status/1';
+      result = statusTypeMatches({credential});
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(TypeError);
+    err.message.should.contain('"@context" must be an array');
+  });
+
+  it('should fail to test the first "@context" value correctly in ' +
+    'statusTypeMatches"', async () => {
+    const id = 'https://example.com/status/1';
+    const list = await createList({length: 100000});
+    const credential = await createCredential({id, list});
+    let err;
+    let result;
+    try {
+      // Changing the @context property intentionally
+      credential['@context'][0] = 'https://example.com/test/1';
+      result = statusTypeMatches({credential});
+    } catch(e) {
+      err = e;
+    }
+    should.exist(err);
+    should.not.exist(result);
+    err.should.be.instanceof(Error);
+    err.message.should.contain('first "@context" value');
+  });
+
+  it('should fail to verify if @context is not an array in ' +
+    '"assertRevocationList2020Context"', async () => {
+    const id = 'https://example.com/status/1';
+    const list = await createList({length: 100000});
+    const credential = await createCredential({id, list});
+    let err;
+    let result;
+    try {
+      // Changing the @context property to a string
```

Comment:
Please use imperative language in comments about what imperative code does (here and elsewhere) -- it also tends to shorten the comments a bit.

```suggestion
      // change the @context property to a string
```